node/src/types/block.rs

```diff
@@ -743,36 +717,98 @@ impl FromBytes for BlockHeader {
         let (parent_hash, remainder) = BlockHash::from_bytes(bytes)?;
         let (state_root_hash, remainder) = Digest::from_bytes(remainder)?;
         let (body_hash, remainder) = Digest::from_bytes(remainder)?;
-        let (deploy_hashes, remainder) = Vec::<DeployHash>::from_bytes(remainder)?;
-        let (transfer_hashes, remainder) = Vec::<DeployHash>::from_bytes(remainder)?;
         let (random_bit, remainder) = bool::from_bytes(remainder)?;
         let (accumulated_seed, remainder) = Digest::from_bytes(remainder)?;
         let (era_end, remainder) = Option::<EraEnd>::from_bytes(remainder)?;
         let (timestamp, remainder) = Timestamp::from_bytes(remainder)?;
         let (era_id, remainder) = EraId::from_bytes(remainder)?;
         let (height, remainder) = u64::from_bytes(remainder)?;
-        let (proposer, remainder) = PublicKey::from_bytes(remainder)?;
         let (next_era_validator_weights, remainder) =
             Option::<BTreeMap<PublicKey, U512>>::from_bytes(remainder)?;
         let block_header = BlockHeader {
             parent_hash,
             state_root_hash,
             body_hash,
-            deploy_hashes,
-            transfer_hashes,
             random_bit,
             accumulated_seed,
             era_end,
             timestamp,
             era_id,
             height,
-            proposer,
             next_era_validator_weights,
         };
         Ok((block_header, remainder))
     }
 }
 
+/// The body portion of a block.
+#[derive(Clone, DataSize, Ord, PartialOrd, Eq, PartialEq, Hash, Serialize, Deserialize, Debug)]
+pub struct BlockBody {
+    proposer: PublicKey,
+    deploy_hashes: Vec<DeployHash>,
+    transfer_hashes: Vec<DeployHash>,
+}
+
+impl BlockBody {
+    /// Create a new body from deploy and transfer hashes.
```

Comment:
```suggestion
    /// Creates a new body from deploy and transfer hashes.
```
(Method doc comments are usually in the third person; also below.)